paperqa/types.py

```diff
@@ -40,17 +40,27 @@
 logger = logging.getLogger(__name__)
 
 
+VAR_MATCH_LOOKUP: Collection[str] = {"1", "true"}
+VAR_MISMATCH_LOOKUP: Collection[str] = {"0", "false"}
+
+DEFAULT_FIELDS_TO_OVERWRITE_FROM_METADATA: set[str] = {
```

Comment:
Let's type hint `Collection[str]` so it's read-only

Reply:
To do that it's needed to set

`model_config = ConfigDict(extra="forbid", arbitrary_types_allowed=True)`

or else I get 

```
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for collections.abc.Collection[str]. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.
```

Is that ok? Or we leave it as a `set`

Reply:
Then you'll need to do:

```python
default_factory=lambda: set(DEFAULT_FIELDS_TO_OVERWRITE_FROM_METADATA)
```